run_custom_lint.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+
+from pathlib import Path
+import typing as T
+
+root = Path(__file__).absolute().parent
+mesonbuild = root / 'mesonbuild'
+
+whitelist = ['mesonbuild/', 'run_', 'ci/', 'tools/', 'docs/']
+
+def check_missing_encoding(lines: T.List[str], path: str) -> int:
+    errors = 0
+    functions = ['read_text', 'write_text', 'open']
+    for num, line in enumerate(lines):
+        for func in functions:
+            l = line
+
+            # Skip ignored lines
+            if '[ignore encoding]' in l:
+                continue
```

Comment:
Why do we need magic comments specific to this script, in order to flag allowed uses of encoding='something other than utf-8'?

If something intentionally other than utf-8 is being used, then it's obviously intentional and therefore obviously fine (at least, ***it will behave the same on Linux and Windows***). So we should just permit it. No need to later on check that the encoding is specifically either 'ascii' or 'utf-8'.

Reply:
Fine, I removed the special cases for `'ascii'` and `'utf-8'`, but this is still needed for stuff like `tarfile.open(`.